qiita_db/support_files/patches/plugin/deblur_v1.0.2_to_v1.0.3_102917.sql

```diff
@@ -0,0 +1,41 @@
+-- With the QIIMEq2 release, the parameters for the command "Pick closed-reference
+-- OTUs" changed. The parameter "reference" has ben substituted for 2 parameters
+-- "reference-tax" and "reference-seq" which contains the path to the actual
+-- reference taxonomy file and the reference sequence file
+
+DO $do$
+DECLARE
+    old_deblur_id BIGINT;
+    new_deblur_id BIGINT;
+    a_info        RECORD;
+    pj_id         UUID;
+BEGIN
+    SELECT command_id INTO old_deblur_id
+        FROM qiita.software_command sc
+            JOIN qiita.software s USING (software_id)
+        WHERE s.name = 'deblur' AND s.version = '1.0.2' AND sc.name = 'deblur-workflow';
+
+    SELECT command_id INTO new_deblur_id
+        FROM qiita.software_command sc
+            JOIN qiita.software s USING (software_id)
+        WHERE s.name = 'deblur' AND s.version = '1.0.3' AND sc.name = 'deblur-workflow';
+
+    -- Modify those artifacts and jobs that have been picked against Greengenes
+    FOR a_info IN
+        SELECT * FROM qiita.artifact WHERE command_id = old_deblur_id
+    LOOP
+        SELECT processing_job_id INTO pj_id
+            FROM qiita.processing_job
+                JOIN qiita.artifact_output_processing_job USING (processing_job_id)
+            WHERE artifact_id = a_info.artifact_id;
+
+        UPDATE qiita.processing_job
+            SET command_parameters = parameters, command_id = new_deblur_id
```

Comment:
`parameters` is not defined. Do parameters need to be changed or they have the same key/value pairs?

Reply:
Nop, parameters not need to be changed:
```python
# moving artifacts from deblur -> deblur
old_d = [c for c in  Software.from_name_and_version('deblur', '1.0.2').commands 
          if c.name == 'deblur-workflow'][0] 
new_d = [c for c in  Software.from_name_and_version('deblur', '1.0.3').commands 
          if c.name == 'deblur-workflow'][0]
          
In [7]: old_d.parameters
Out[7]: 
{'error-dist': ['string',
  '1, 0.06, 0.02, 0.02, 0.01, 0.005, 0.005, 0.005, 0.001, 0.001, 0.001, 0.0005'],
 'indel-max': ['integer', '3'],
 'indel-prob': ['float', '0.01'],
 'jobs-to-start': ['integer', '1'],
 'mean-error': ['float', '0.005'],
 'min-reads': ['integer', '0'],
 'min-size': ['integer', '2'],
 'neg-ref-db-fp': ['choice:["default"]', 'default'],
 'neg-ref-fp': ['choice:["default"]', 'default'],
 'pos-ref-db-fp': ['choice:["default"]', 'default'],
 'pos-ref-fp': ['choice:["default"]', 'default'],
 'seqs-fp': ['artifact', None],
 'threads-per-sample': ['integer', '1'],
 'trim-length': ['integer', '-1']}

In [8]: new_d.parameters
Out[8]: 
{'error-dist': ['string',
  '1, 0.06, 0.02, 0.02, 0.01, 0.005, 0.005, 0.005, 0.001, 0.001, 0.001, 0.0005'],
 'indel-max': ['integer', '3'],
 'indel-prob': ['float', '0.01'],
 'jobs-to-start': ['integer', '1'],
 'mean-error': ['float', '0.005'],
 'min-reads': ['integer', '0'],
 'min-size': ['integer', '2'],
 'neg-ref-db-fp': ['choice:["default"]', 'default'],
 'neg-ref-fp': ['choice:["default"]', 'default'],
 'pos-ref-db-fp': ['choice:["default"]', 'default'],
 'pos-ref-fp': ['choice:["default"]', 'default'],
 'seqs-fp': ['artifact', None],
 'threads-per-sample': ['integer', '1'],
 'trim-length': ['integer', '-1']}
```